mindsdb/integrations/handlers/s3_handler/s3_handler.py

```diff
@@ -17,129 +15,198 @@
     RESPONSE_TYPE
 )
 
+from mindsdb.integrations.libs.base import DatabaseHandler
+
 
 logger = log.getLogger(__name__)
 
+
 class S3Handler(DatabaseHandler):
     """
-    This handler handles connection and execution of the S3 statements.
+    This handler handles connection and execution of the SQL statements on AWS S3.
     """
 
     name = 's3'
+    table_name = 's3_table'
+    # TODO: Can other file formats be supported?
+    supported_file_formats = ['csv', 'tsv', 'json', 'parquet']
 
-    def __init__(self, name: str, connection_data: Optional[dict], **kwargs):
+    def __init__(self, name: Text, connection_data: Optional[Dict], **kwargs):
         """
-        Initialize the handler.
+        Initializes the handler.
+
         Args:
-            name (str): name of particular handler instance
-            connection_data (dict): parameters for connecting to the database
-            **kwargs: arbitrary keyword arguments.
+            name (Text): The name of the handler instance.
+            connection_data (Dict): The connection data required to connect to the AWS (S3) account.
+            kwargs: Arbitrary keyword arguments.
         """
         super().__init__(name)
-        self.parser = parse_sql
-        self.dialect = 's3'
         self.connection_data = connection_data
         self.kwargs = kwargs
+        self.key = None
 
         self.connection = None
         self.is_connected = False
 
+        self.is_select_query = False
+        self.key = None
+
     def __del__(self):
         if self.is_connected is True:
             self.disconnect()
 
-    def connect(self) -> StatusResponse:
+    def connect(self) -> DuckDBPyConnection:
         """
-        Set up the connection required by the handler.
+        Establishes a connection to the AWS (S3) account via DuckDB.
+
+        Raises:
+            KeyError: If the required connection parameters are not provided.
+
         Returns:
-            HandlerStatusResponse
+            boto3.client: A client object to the AWS (S3) account.
         """
-
         if self.is_connected is True:
             return self.connection
 
-        self.connection = boto3.client(
-            's3',
-            aws_access_key_id=self.connection_data['aws_access_key_id'],
-            aws_secret_access_key=self.connection_data['aws_secret_access_key'],
-            region_name=self.connection_data['region_name']
-        )
+        # Validate mandatory parameters.
+        if not all(key in self.connection_data for key in ['aws_access_key_id', 'aws_secret_access_key', 'bucket']):
+            raise ValueError('Required parameters (aws_access_key_id, aws_secret_access_key, bucket) must be provided.')
+
+        # Connect to S3 via DuckDB and configure mandatory credentials.
+        self.connection = self._connect_duckdb()
```

Comment:
We may catch exception here or in the connect_duckdb bellow

Reply:
@ZoranPandovski I noticed that even if we provide incorrect credentials, these `SET` commands run without an issue. From what I have understood, the credentials are checked only when running a query against the S3 bucket. I can add some generic error handling around this. Do you think it is necessary?